lib/plugins/create/create.js

```diff
@@ -225,49 +226,40 @@ class Create {
       this.serverless.config.update({ servicePath: process.cwd() });
     }
 
-    // copy template files recursively to cwd
-    // while keeping template file tree
-    try {
-      this.serverless.utils.copyDirContentsSync(templateSrcDir, process.cwd());
+    return createFromTemplate(this.options.template, process.cwd()).then(
+      () => {
+        // rename the service if the user has provided a path via options and is creating a service
+        if ((boilerplatePath || serviceName) && notPlugin) {
+          const newServiceName = serviceName || boilerplatePath.split(path.sep).pop();
 
-      // NPM renames .gitignore to .npmignore on publish so we have to rename it.
-      if (fse.existsSync(path.join(process.cwd(), 'gitignore'))) {
-        fse.renameSync(
-          path.join(process.cwd(), 'gitignore'),
-          path.join(process.cwd(), '.gitignore')
-        );
-      }
-    } catch (err) {
-      const errorMessage = [
-        'Error unable to create a service in this directory. ',
-        'Please check that you have the required permissions to write to the directory',
-      ].join('');
-
-      throw new this.serverless.classes.Error(errorMessage);
-    }
+          renameService(newServiceName, this.serverless.config.servicePath);
+        }
 
-    // rename the service if the user has provided a path via options and is creating a service
-    if ((boilerplatePath || serviceName) && notPlugin) {
-      const newServiceName = serviceName || boilerplatePath.split(path.sep).pop();
+        userStats.track('service_created', {
```

Comment:
Oprhaned promise, but I guess this is still something we do in such situations.

Reply:
Technically it's left the way it was (change is due to refactor that relocated that part in file).

Yes, we do it like that now. Idea is not bad, but the way it's implemented it's prone to issues (especially that we shortcut process with `process.exit`, so there's no guarantee those requests will fulfill, but it's not a concern for this PR - and we discuss it already in other channel)